vividus-facade-jira/src/main/java/org/vividus/jira/JiraClientProvider.java

```
@@ -0,0 +1,148 @@
+            {
+                return getDefaultOrElseThrow(() -> error("Issue key is not specified"));
+            }
+
+            Set<Entry<String, JiraConfiguration>> configurations = jiraConfigurations.getData().entrySet().stream()
+                    .filter(cnf -> cnf.getValue().getProjectCodePattern().matcher(issueKey).matches())
+                    .collect(Collectors.toSet());
+
+            if (configurations.size() == 1)
+            {
+                return getByFirst(configurations);
+            }
+            else if (configurations.size() > 1)
+            {
+                throw error("More than one JIRA configuration maps to %s code: %s", issueKey,
+                        configurations.stream().map(Entry::getKey).collect(Collectors.toList()));
+            }
+
+            return getDefaultOrElseThrow(
+                () -> error("Unable to find configuration for JIRA project code %s", issueKey));
+        }
+        catch (ExecutionException thrown)
+        {
+            throw new JiraConfigurationException(thrown.getCause());
+        }
+    }
+
+    private JiraClient getDefaultOrElseThrow(Supplier<JiraConfigurationException> exceptionSupplier)
+            throws ExecutionException, JiraConfigurationException
+    {
+        Set<Entry<String, JiraConfiguration>> configurations = jiraConfigurations.getData().entrySet();
+        if (configurations.size() == 1)
+        {
+            return getByFirst(configurations);
+        }
+
+        throw exceptionSupplier.get();
+    }
+
+    private JiraClient getByFirst(Set<Entry<String, JiraConfiguration>> configurations) throws ExecutionException
+    {
+        return jiraClients.get(configurations.iterator().next().getValue());
+    }
+
+    private JiraConfigurationException error(String message, Object... args)
+    {
+        return new JiraConfigurationException(String.format(message, args));
+    }
+
+    public static class JiraConfigurationException extends Exception
```

```suggestion
    public static class JiraConfigurationException extends RuntimeException
```

---

why runtime? I remember we had conversation on this and decided to make it checked

---

even though it seems that there is no reason to proceed with import process with this kind of exception, the end user may want in the future to configure several instances to export to several xray at the same time, so we should not interrupt the process

---

also if we add this for known issue we will have to catch runtime exception to not broke tests flow

---

> why runtime? I remember we had conversation on this and decided to make it checked

you never catch it

---

> also if we add this for known issue we will have to catch runtime exception to not broke tests flow

and this case you'll make it non-runtime, but most likely you'll catch all exceptions

---

I catch it in XrayExporter 154

---

I'm not sure it should be there at all